plugins/osdn/flannelmt/bin/openshift-sdn-flannelmt

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+set -e
+
+lock_file=/var/lock/openshift-sdn.lock
+
+action=$1
+pod_namespace=$2
+pod_name=$3
+net_container=$4
+tenant_id=$5
+
+lockwrap() {
+    (
+    flock 200
+    "$@"
+    ) 200>${lock_file}
+}
+
+# Retrieve the name of the host-local member of the veth pair that
+# connects the container (identified by pid) to the docker bridge.
+get_veth_host() {
+    local pid=$1
+
+    local veth_ifindex=$(nsenter -n -t $pid -- ethtool -S eth0 | sed -n -e 's/.*peer_ifindex: //p')
+    # Strip a suffix starting with '@' from the interface name.
+    # The suffixed interface name won't be recognized by brctl or ovs-*
+    ip link show | sed -ne "s/^$veth_ifindex: \([^:@]*\).*/\1/p"
+}
+
+get_ipaddr_pid_veth() {
+    network_mode=$(docker inspect --format "{{.HostConfig.NetworkMode}}" ${net_container})
+    if [ "${network_mode}" == "host" ]; then
+      # quit, nothing for the SDN here
+      exit 0
+    elif [[ "${network_mode}" =~ container:.* ]]; then
+      # Get pod infra container
+      net_container=$(echo ${network_mode} | cut -d ":" -f 2)
+    fi
+    ipaddr=$(docker inspect --format "{{.NetworkSettings.IPAddress}}" ${net_container})
+    pid=$(docker inspect --format "{{.State.Pid}}" ${net_container})
+    veth_host=$(get_veth_host $pid)
+}
+
+add_ovs_port() {
+    brctl delif lbr0 $veth_host
+    ovs-vsctl add-port br0 ${veth_host}
+}
+
+del_ovs_port() {
+    ovs-vsctl --if-exists del-port $veth_host
+}
+
+add_ovs_flows() {
+    ovs_port=$(ovs-ofctl -O OpenFlow13 dump-ports-desc br0  | grep ${veth_host} | cut -d "(" -f 1 | tr -d ' ')
+
+    ovs-ofctl -O OpenFlow13 add-flow br0 "table=3,priority=100,in_port=${ovs_port},ip,nw_src=${ipaddr},actions=load:${tenant_id}->NXM_NX_REG0[],goto_table:4"
```

Comment:
This isn't going to work; you're making openshift-sdn depend on the precise details of how flannel sets up OVS, so this locks you to a single specific version of flannel. We have a hard enough time keeping openshift-sdn and origin in sync; there's no way we can try to stay in sync with an external module.

Flannel needs to install some script that we can call here that abstracts out these details.